docs/site/Controllers.md

```diff
@@ -122,104 +148,125 @@ The `@api` decorator allows you to annotate your Controller with a specification
-
-Listed below are some of the most common error codes. The full list of supported codes is found [here](https://github.com/jshttp/http-errors#list-of-all-constructors).
-
-|Status Code|Error                |
-|-----------|---------------------|
-|400        |BadRequest           |
-|401        |Unauthorized         |
-|403        |Forbidden            |
-|404        |NotFound             |
-|500        |InternalServerError  |
-|502        |BadGateway           |
-|503        |ServiceUnavailable   |
-|504        |GatewayTimeout       |
-
-The example below shows the previous controller revamped with `HttpErrors` along with a test to verify that the error is thrown properly.
+In order to specify errors for controller methods to throw, the class
+`HttpErrors` is used. `HttpErrors` is a class that has been re-exported from
+[http-errors](https://www.npmjs.com/package/http-errors), and can be found in
+the `@loopback/rest` package.
+
+Listed below are some of the most common error codes. The full list of supported
+codes is found
+[here](https://github.com/jshttp/http-errors#list-of-all-constructors).
+
+| Status Code | Error               |
+| ----------- | ------------------- |
+| 400         | BadRequest          |
+| 401         | Unauthorized        |
+| 403         | Forbidden           |
+| 404         | NotFound            |
+| 500         | InternalServerError |
+| 502         | BadGateway          |
+| 503         | ServiceUnavailable  |
+| 504         | GatewayTimeout      |
+
+The example below shows the previous controller revamped with `HttpErrors` along
+with a test to verify that the error is thrown properly.
 
 ```js
 // the test
-import {HelloController} from 'path.to.controller';
-import {HttpErrors, expect} from '@loopback/testlab';
+import {HelloController} from '../../../src/controllers';
+import {HelloRepository} from '../../../src/repositories';
+import {testdb} from '../../fixtures/datasources/testdb.datasource';
+import {expect} from '@loopback/testlab';
+import {HttpErrors} from '@loopback/rest';
+
+const HttpError = HttpErrors.HttpError;
 
 describe('Hello Controller', () => {
-  it('returns 422 Unprocessable Entity for non-positive limit', async () => {
-    const controller = new HelloController();
-    let errCaught: Error;
-    try {
-      await controller.list(0.4); // an HttpError should be thrown here
-    } catch (err) {
-      errCaught = err;
-    }
-    // the test fails here if the error was not thrown
-    expect(errCaught).to.have.property('statusCode', 422);
-    expect(errCaught.message).to.match(/non-positive/i);
+  it('returns 422 Unprocessable Entity for non-positive limit', () => {
+    const repo = new HelloRepository(testdb);
+    const controller = new HelloController(repo);
+
+    return expect(controller.list(0.4)).to.be.rejectedWith(HttpError, {
+      message: 'limit is non-positive',
+      statusCode: 422,
+    });
   });
 });
+
 ```
+
 ```js
 // the controller
```

Comment:
Shouldn't the controller go before the test?

Reply:
Show file name as a comment

Reply:
TDD advocates that the test be written first. I remember writing this section some time ago, and @bajtos left a review comment to have the test show up first